File: schemas/inventory_achievement_association.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class InventoryAchievementAssociationSchemaBase(BaseModel):
    id: int
    inventory_id: int
    achievement_id: int
    comment: Optional[str]
    added_at: datetime

    model_config = {
        "from_attributes": True
    }


class InventoryAchievementAssociationSchemaFull(InventoryAchievementAssociationSchemaBase):
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True
    }

File: frontend2/types/inventoryAchievementAssociation.ts
export interface InventoryAchievementAssociationSchemaBase {
  id: number;
  inventoryId: number;
  achievementId: number;
  comment?: string | null;
  addedAt: string; // ISO date string
}

export interface InventoryAchievementAssociationSchemaFull extends InventoryAchievementAssociationSchemaBase {
  createdAt: string;
  updatedAt: string;
}
